### PR TITLE
fix sql transaction error in statistic API

### PR DESCRIPTION
**Pull request by waltcow**
https://github.com/langgenius/dify/issues/1583

### CODE REVIEW

File: api/controllers/console/app/statistic.py [hunk 1]
```diff
@@ -65,13 +65,13 @@ def get(self, app_id):
         with db.engine.begin() as conn:
             rs = conn.execute(db.text(sql_query), arg_dict)
 
-        response_data = []
+            response_data = []
```
Comment (takatost) [inline api/controllers/console/app/statistic.py]: Defining `response_data` inside the `with` statement might cause issues when trying to access it in the final `return` statement. Put it above `with` would be better.

Comment (takatost) [reply api/controllers/console/app/statistic.py]: The changes below are the same, you know. 
Whatever, thank you very much for your contribution! 😊

**Review by takatost — CHANGES_REQUESTED** [review]

File: api/controllers/console/app/statistic.py (hunk 1 above)
Comment (waltcow) [reply api/controllers/console/app/statistic.py]: Fixed @takatost

**Review by takatost — APPROVED** [review]
thx!

**Pull request merged**